nodenetworkconfig/api/v1alpha/nodenetworkconfig.go

```diff
@@ -0,0 +1,81 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Important: Run "make" to regenerate code after modifying this file
+
+// +kubebuilder:object:root=true
+
+// NodeNetworkConfig is the Schema for the nodenetworkconfigs API
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:subresource:status
+type NodeNetworkConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NodeNetworkConfigSpec   `json:"spec,omitempty"`
+	Status NodeNetworkConfigStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// NodeNetworkConfigList contains a list of NetworkConfig
+type NodeNetworkConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []NodeNetworkConfig `json:"items"`
+}
+
+// NodeNetworkConfigSpec defines the desired state of NetworkConfig
+type NodeNetworkConfigSpec struct {
+	RequestedIPCount int64    `json:"requestedIPCount,omitempty"`
+	IPsNotInUse      []string `json:"iPsNotInUse,omitempty"`
+}
+
+// NodeNetworkConfigStatus defines the observed state of NetworkConfig
+type NodeNetworkConfigStatus struct {
+	AssignedIPCount         int64              `json:"assignedIPCount,omitempty"`
```

Comment:
the assigned up count here and in network container seem unnecessary can just len() IpAssignments.  It's a slice not a stream :)

Reply:
@neaggarwMS wanted these for easier debugging/ops. If you don't object @paulgmiller, I'll leave them in. If you do, we can discuss with Neha.